nuxeo-core/nuxeo-core-kv/src/main/java/org/nuxeo/ecm/core/kv/AbstractKeyValueStoreProvider.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * (C) Copyright 2017 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Florent Guillaume
+ */
+package org.nuxeo.ecm.core.kv;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.CharacterCodingException;
+import java.nio.charset.CharsetDecoder;
+import java.nio.charset.CodingErrorAction;
+
+/**
+ * Key/Value Store common methods.
+ *
+ * @since 9.3
+ */
+public abstract class AbstractKeyValueStoreProvider implements KeyValueStoreProvider {
+
+    protected static final ThreadLocal<CharsetDecoder> UTF_8_DECODERS = ThreadLocal.withInitial(
```

Comment:
`UTF_8_DECODER` as there is only one, no?

Reply:
I used the plural because it's a thread-local that actually contains one decoder per thread, so there are several. I see what you mean but I don't know if there's a best practice here.